pinot-core/src/main/java/org/apache/pinot/core/transport/grpc/GrpcQueryServer.java

```diff
@@ -48,11 +51,14 @@
   private final Server _server;
   private final ExecutorService _executorService =
       Executors.newFixedThreadPool(ResourceManager.DEFAULT_QUERY_WORKER_THREADS);
+  private final AccessControl _accessControl;
 
-  public GrpcQueryServer(int port, QueryExecutor queryExecutor, ServerMetrics serverMetrics) {
+  public GrpcQueryServer(int port, QueryExecutor queryExecutor, ServerMetrics serverMetrics,
+      AccessControlFactory accessControlFactory) {
     _queryExecutor = queryExecutor;
     _serverMetrics = serverMetrics;
     _server = ServerBuilder.forPort(port).addService(this).build();
+    _accessControl = accessControlFactory.create();
```

Comment:
Why not inject the `AccessControl` instead of the factory to make this more testable?

Reply:
Was following the old behavior to pass the Factory, but I think you are right, we should create the ACL in ServerInstance then set it to both nettyTlsQueryServer and grpcQueryServer.